src/features/todos.ts

```diff
@@ -1,9 +1,23 @@
 import { Todo } from '../types/Todo';
 
-export const actions = {};
+type AddAction = { type: 'todos/ADD'; payload: Todo };
 
-const todosReducer = (): Todo[] => {
-  return [];
+const addTodos = (todo: Todo): AddAction => ({
+  type: 'todos/ADD',
+  payload: todo,
+});
+
+export const actions = { addTodos };
+
+// eslint-disable-next-line @typescript-eslint/default-param-last
+const todosReducer = (todos: Todo[] = [], action: AddAction): Todo[] => {
+  switch (action.type) {
+    case 'todos/ADD':
+      return [...todos, action.payload];
```

Comment:
since in this case, you don't add a single todo actually it would be more effective to add all todos in one go. Pass the array of todos as payload and add them all
```suggestion
      return [...todos, ...action.payload];
```
P.S. change the action name accordingly